.github/workflows/conda-ci.yml

```diff
@@ -0,0 +1,72 @@
+name: C++ CI Workflow with dependencies installed via conda
+
+on:
+  push:
+  pull_request:
+  schedule:
+  # * is a special character in YAML so you have to quote this string
+  # Execute a "nightly" build at 2 AM UTC 
+  - cron:  '0 2 * * *'
+
+jobs:
+  build:
+    name: '[${{ matrix.os }}@${{ matrix.build_type }}@conda]'
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        build_type: [Release]
+        os: [ubuntu-latest, windows-latest, macOS-latest]
+      fail-fast: false
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - uses: conda-incubator/setup-miniconda@v2
+      with:
+        mamba-version: "*"
+        channels: conda-forge,robotology
+        channel-priority: true
+
+    - name: Dependencies
+      shell: bash -l {0}
+      run: |
+        # Compilation related dependencies 
+        mamba install cmake compilers ninja pkg-config
+        # Actual dependencies
+        mamba install eigen blockfactory idyntree osqp-eigen qpoases yarp
+        
+    - name: Dependencies
+      shell: bash -l {0}
+      run: |
+        # Compilation related dependencies 
+        mamba install cmake compilers ninja pkg-config
+        # Actual dependencies
+        mamba install eigen blockfactory idyntree osqp-eigen qpoases yarp
+
+    - name: Configure [Linux&macOS]
+      if: contains(matrix.os, 'macos') || contains(matrix.os, 'ubuntu')
+      shell: bash -l {0}
+      run: |
+        mkdir -p build
+        cd build
+        cmake -GNinja -DBUILD_TESTING:BOOL=ON -DCMAKE_BUILD_TYPE=${{ matrix.build_type }} ..
+      
+    - name: Configure [Windows]
+      if: contains(matrix.os, 'windows')
+      shell: bash -l {0}
+      run: |
+        mkdir -p build
+        cd build
+        cmake -G"Visual Studio 16 2019" -DBUILD_TESTING:BOOL=ON -DCMAKE_BUILD_TYPE=${{ matrix.build_type }} ..
+
+    - name: Build 
+      shell: bash -l {0}
+      run: |
+        cd build
+        cmake --build . --config ${{ matrix.build_type }}
+        
+    - name: Test
+      shell: bash -l {0}
+      run: |
+        cd build
+        ctest --output-on-failure -C ${{ matrix.build_type }}
```

Comment:
I don't think we have tests here, maybe we can leave these lines commented out?

Reply:
This line works fine also for repo that do not have tests, and will automatically start to run tests if tests are added. Given that I plan to wrap this logic in a github action to avoid duplicate code, and I would keep this for consistency with the rest of other repos.